services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/tenant/ManagementStore.java

```diff
@@ -0,0 +1,422 @@
+
+    /**
+     * Delete the tenant.
+     *
+     * @param tenantId The tenant to delete.
+     * @param resourceVersion The version of the resource to delete.
+     * @param spanContext The span to contribute to.
+     * @return The future, tracking the outcome of the operation.
+     */
+    public Future<UpdateResult> delete(final String tenantId, final Optional<String> resourceVersion, final SpanContext spanContext) {
+
+        final Span span = TracingHelper.buildChildSpan(this.tracer, spanContext, "delete tenant", getClass().getSimpleName())
+                .withTag(TracingHelper.TAG_TENANT_ID, tenantId)
+                .start();
+
+        resourceVersion.ifPresent(version -> span.setTag("version", version));
+
+        final Statement statement;
+        if (resourceVersion.isPresent()) {
+            statement = this.deleteVersionedStatement;
+        } else {
+            statement = this.deleteStatement;
+        }
+
+        final var expanded = statement.expand(map -> {
+            map.put("tenant_id", tenantId);
+            resourceVersion.ifPresent(version -> map.put("expected_version", version));
+        });
+
+        log.debug("delete - statement: {}", expanded);
+
+        final var result = expanded
+                .trace(this.tracer, span.context())
+                .update(this.client)
+                .onComplete(x -> span.finish());
+
+        return checkOptimisticLock(
+                result, span,
+                resourceVersion,
+                checkSpan -> readTenantEntryById(this.client, tenantId, checkSpan.context()))
+                .onComplete(x -> span.finish());
+
+    }
+
+    /**
+     * Create a new tenant.
+     * <p>
+     * The operation may fail with a {@link org.eclipse.hono.service.base.jdbc.store.EntityNotFoundException} if the
+     * specified tenant does not exist.
+     * <p>
+     * The operation may fail with a {@link org.eclipse.hono.service.base.jdbc.store.DuplicateKeyException} if a
+     * tenant with the ID or trust anchor already exists.
+     * <p>
+     * The operation may fail with an {@link org.eclipse.hono.service.base.jdbc.store.OptimisticLockingException} if
+     * an expected resource version was provided, but the current version did not match.
+     *
+     * @param tenantId The ID of the new tenant.
+     * @param tenant The tenant information.
+     * @param resourceVersion An optional resource version.
+     * @param spanContext The span to contribute to.
+     * @return A future, tracking the outcome of the operation.
+     */
+    public Future<Versioned<Void>> update(final String tenantId, final Tenant tenant, final Optional<String> resourceVersion, final SpanContext spanContext) {
+
+        final var json = tenantToJson(tenant);
+
+        final Span span = TracingHelper.buildChildSpan(this.tracer, spanContext, "update tenant", getClass().getSimpleName())
+                .withTag(TracingHelper.TAG_TENANT_ID, tenantId)
+                .withTag("data", json)
```

Comment:
Do you intend to provide full text searching on all CRUD method payloads via the tracing system? We do nothing like this for the other device registries so I wonder why we would want/need to do this in the JDBC registry.
FMPOV the Tags are intended as markers for the spans by which it is easy to group and filter them. Adding full payloads here doesn't seem right to me. If at all, the payload should be logged to the span but it should not go into a tag ....

Reply:
Is there anything else on this PR to fix?

Reply:
As I wrote above, FMPOV the data should not be written into Tags ...

Reply:
Yes, I understand that. But with every round of changes I add, new comments come up. Instead of endlessly updating the PR, I would like to understand "what else" I need to change on the PR, in order get it merged.

Reply:
Sorry for the inconvenience. Nothing else from my side.